src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -12372,6 +12378,9 @@ CorJitResult CallCompileMethodWithSEHWrapper(EEJitManager *jitMgr,
          }
     }
 
+    if (ftn->HasNativeCallableAttribute())
```

Comment:
This will also need to be duplicated for crossgen2.

Reply:
I believe this is already done.

https://github.com/dotnet/runtime/blob/b0d39e472ae1ce615818f6d8bcc78597e3d2eac4/src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs#L2838-L2839

Reply:
Interesting. But I doubt that it is working end-to-end. 

We may also need to add `READYTORUN_HELPER_...` for the two new helpers to make it actually work.

Reply:
Also, crossgen2 should get the same checks for invalid use as the runtime.

Reply:
I'm not entirely sure where these checks should go. Can you point out an example of when this check would be possible?

Reply:
I am not sure either. @MichalStrehovsky or @trylek should have an opinion.

Reply:
> Also, crossgen2 should get the same checks for invalid use as the runtime.

I wonder if it would be better to place the validation in `getCallInfo` in both JitInterface.cpp and Crossgen2.

`getFunctionFixedEntrypoint` is usually not called when crossgenning:

```cpp
#ifdef FEATURE_READYTORUN_COMPILER
        if (tree->AsFptrVal()->gtEntryPoint.addr != nullptr)
        {
            addrInfo = tree->AsFptrVal()->gtEntryPoint;
        }
        else
#endif
        {
            info.compCompHnd->getFunctionFixedEntryPoint(tree->AsFptrVal()->gtFptrMethod, &addrInfo);
        }
```

Reply:
@MichalStrehovsky Would you expect the `getCallInfo` to throw an exception/fail? I am unclear what information I would pass back in the `getCallInfo` call. Something like `invalidLdftnCall`? Who would be expected to respond to that?

Reply:
> Would you expect the getCallInfo to throw an exception/fail

Yes

Reply:
We should be able to throw from there. E.g. here we already do:

https://github.com/dotnet/runtime/blob/71deb1eea5c1f6866833f06ee5f295106b6e24b6/src/coreclr/src/vm/jitinterface.cpp#L5083-L5089